cpp/src/arrow/flight/test_definitions.cc

```diff
@@ -854,8 +854,8 @@ Status AppMetadataTestServer::DoGet(const ServerCallContext& context,
     RETURN_NOT_OK(ExampleIntBatches(&batches));
   }
   ARROW_ASSIGN_OR_RAISE(auto batch_reader, RecordBatchReader::Make(batches));
-  *data_stream = std::unique_ptr<FlightDataStream>(new NumberingStream(
-      std::unique_ptr<FlightDataStream>(new RecordBatchStream(batch_reader))));
+  *data_stream = std::make_unique<FlightDataStream>(
+      NumberingStream(std::make_unique<RecordBatchStream>(batch_reader)));
```

Comment:
Can we use `std::make_unique<NumberingStream>(std::make_unique<RecordBatchStream>(batch_reader))` here?

Reply:
done